bitnami/grafana-alloy/templates/application.yaml

```diff
@@ -0,0 +1,299 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if eq .Values.alloy.resourceType "deployment" }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: Deployment
+{{- else if eq .Values.alloy.resourceType "statefulset" }}
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
+kind: StatefulSet
+{{- else }}
+apiVersion: {{ include "common.capabilities.daemonset.apiVersion" . }}
+kind: DaemonSet
+{{- end }}
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: alloy
+    app.kubernetes.io/part-of: alloy
+  {{- if or .Values.daemonsetAnnotations .Values.deploymentAnnotations .Values.statefulsetAnnotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" (dict "values" (list .Values.daemonsetAnnotations .Values.deploymentAnnotations .Values.statefulsetAnnotations .Values.commonAnnotations) "context" .) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if and .Values.autoscaling.hpa.enabled (not (eq .Values.alloy.resourceType "daemonset")) }}
+  replicas: {{ .Values.alloy.replicaCount }}
+  {{- end }}
+  {{- if eq .Values.alloy.resourceType "statefulset" }}
+  podManagementPolicy: {{ .Values.podManagementPolicy | quote }}
+  {{- end }}
+  {{- if .Values.updateStrategy }}
+  {{- if eq .Values.alloy.resourceType "deployment" }}
+  strategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
+  {{- else if eq .Values.alloy.resourceType "statefulset" }}
+  updateStrategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
+  serviceName: {{ template "common.names.fullname" . }}
+  {{- else }}
+  updateStrategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
+  {{- end }}
+  {{- end }}
+  {{- $podLabels := include "common.tplvalues.merge" (dict "values" (list .Values.podLabels .Values.commonLabels) "context" .) }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" $podLabels "context" $ ) | nindent 6 }}
+      app.kubernetes.io/component: alloy
+      app.kubernetes.io/part-of: alloy
+  template:
+    metadata:
+      {{- if .Values.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
```

Comment:
I'd include "checksum" annotations in case the associated configmap or secret change.

Reply:
This isn't done, right? I'm missing the ConfigMap checksum.